### PR TITLE
chore: test negative timestamps against debit

### PR DESCRIPTION
**Pull request by dcaf-matcha**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     ],
     "packageManager": "yarn@3.6.0",
     "scripts": {
-        "postinstall": "husky install",
```
Comment (dcaf-mocha) [inline package.json:10]: lets add a ticket to add the yarn v3 plugin to replace this, this way other contributors other then us can have an easier time getting the repo setup on their machine.

File: program-tests/pre-authorized-debit-v1/debit/oneTime.shared.ts [hunk 2]
```diff
@@ -423,6 +423,154 @@ export function testOneTimeDebit(
...
+              preAuthorization: preAuthorizationPubkey,
+              tokenProgram: tokenProgramId,
+            })
+            .signers([debitAuthorityKeypair])
+            .rpc(),
+        ).to.eventually.be.rejectedWith(
+          /Error Code: PreAuthorizationNotActive. Error Number: 6000/,
+        );
+      });
+      it("fails if activation is negative but quantitatively larger than past expiry", async () => {
```
Comment (dcaf-mocha) [inline program-tests/pre-authorized-debit-v1/debit/oneTime.shared.ts:471]: Nit: this test title is a little confusing.

**Review by dcaf-mocha — APPROVED** [review]

File: programs/pre-authorized-debit-v1/src/instructions/debit.rs [hunk 3]
```diff
@@ -298,6 +315,98 @@ mod tests {
...
+    #[test_case(5, 4, false, 100, 100, 400, 100)]
+    #[test_case(5, 4, false, 100, 400, 400, 100)]
+    #[test_case(5, 5, false, 100, 100, 100, 400)]
+    #[test_case(5, 5, false, 100, 100, 400, 100)]
+    #[test_case(5, 5, false, 100, 400, 400, 100)]
+    #[test_case(5, 5, false, 100, 0, 500, 0)]
+    #[test_case(5, 5, false, 100, 400, 500, 0)]
+    #[test_case(5, 5, false, 100, 500, 500, 0)]
+    #[test_case(5, 5, false, 100, 0, 100, 400)]
+    // recurring pre-auth (available amount resets every cycle)
```
[the remark this answers is not in the document]
Comment (dcaf-matcha) [reply programs/pre-authorized-debit-v1/src/instructions/debit.rs:355]: done in https://github.com/dcaf-labs/pre-authorized-debit/pull/32/commits/3a503caf892c6dcf874b3c034cd6e06fa2b03eb6

**Review by dcaf-mocha — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (dcaf-matcha) [reply package.json:10]: added a ticket to notion

File: program-tests/pre-authorized-debit-v1/debit/oneTime.shared.ts (hunk 2 above)
Comment (dcaf-matcha) [reply program-tests/pre-authorized-debit-v1/debit/oneTime.shared.ts]: fixed

**Review by dcaf-mocha — APPROVED** [review]

**Pull request merged**